### PR TITLE
Add permissions required to README.md

### PR DESCRIPTION
**Pull request by francorbacho**
This was pretty confusing to me, and it looks like obvious information to provide to new users.

### CODE REVIEW

**Comment by ggrossetie** [comment]
I didn't know it was needed. Were did you configure this setting?

**Comment by francorbacho** [comment]
It was failing for me (at least on private repos) with an authorization error without it. It was all configured in the `job.yml` file like so:

```yaml
name: Foo

on:
  push:
  pull_request:

jobs:
  run-job:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      checks: write
    steps:
      - ...
```

**Comment by ggrossetie** [comment]
I found an explanation at: https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs 

You are probably using a "restricted" `GITHUB_TOKEN` as mentioned in: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

I think we should explain when it's needed and why it's needed. As far as I understand, it's only needed when using a "restricted" `GITHUB_TOKEN`. 

Do you have the authorization error message? It might also help others to troubleshoot if we copy/paste the error message next to the solution.

Thanks!

**Comment by francorbacho** [comment]
I don't know why I have a restricted access `GITHUB_TOKEN`, I don't think I configured it anywhere.

Anyhow, I have included the information you requested. Feel free to rearrange the information as you wish.

File: README.md [hunk 1]
```diff
@@ -18,6 +18,26 @@ In order to use this action, you will need to generate a JSON file using the fol
...
+HttpError: Resource not accessible by integration
+```
+
+You need to provide the `checks` write permission:
+```yaml
+permissions:
+  checks: write
+```
+
+Note that setting a `permissions` field will disable all omitted permissions, like
```
Comment (ggrossetie) [inline README.md:38]:
```suggestion
You can use `permissions` either as a top-level key, to apply to all jobs in the workflow, or within specific jobs. When you add the permissions key within a specific job, all actions and run commands within that job that use the `GITHUB_TOKEN` gain the access rights you specify.

**IMPORTANT**: Setting a `permissions` field will disable all omitted permissions, like
```

File: README.md [hunk 2]
```diff
@@ -18,6 +18,26 @@ In order to use this action, you will need to generate a JSON file using the fol
...
+the `contents: read`, which is needed to clone and read your repository.
```
Comment (ggrossetie) [inline README.md:39]:
```suggestion
the `contents: read`, which is needed to clone and read your repository.

You can learn more about `GITHUB_TOKEN` permissions at: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
```

File: README.md [hunk 3]
```diff
@@ -18,6 +18,30 @@ In order to use this action, you will need to generate a JSON file using the fol
 ]
 ```
 
+## Permissions
+
+It is possible you have a `GITHUB_TOKEN` with restricted access by default, in which
+case you might get the following error:
```
Comment (ggrossetie) [inline README.md:24]:
```suggestion
It is possible you have a `GITHUB_TOKEN` with restricted access by default, in which case you might get the following error:
```

File: README.md [hunk 4]
```diff
@@ -18,6 +18,30 @@ In order to use this action, you will need to generate a JSON file using the fol
...
+
+```
+Error: GitHubApiUnauthorizedError: Unable to create a check, please make sure that
+the provided 'repo-token' has write permissions to 'your/repo' - cause:
+HttpError: Resource not accessible by integration
```
Comment (ggrossetie) [inline README.md:29]:
```suggestion
Error: GitHubApiUnauthorizedError: Unable to create a check, please make sure that the provided 'repo-token' has write permissions to 'your/repo' - cause: HttpError: Resource not accessible by integration
```

File: README.md [hunk 5]
```diff
@@ -18,6 +18,30 @@ In order to use this action, you will need to generate a JSON file using the fol
...
+```
+
+You need to provide the `checks` write permission:
+```yaml
+permissions:
+  checks: write
+```
+
+You can use `permissions` either as a top-level key, to apply to all jobs in the workflow, or within specific jobs. When you add the permissions key within a specific job, all actions and run commands within that job that use the `GITHUB_TOKEN` gain the access rights you specify.
```
Comment (ggrossetie) [inline README.md:38]:
```suggestion
You can use `permissions` either as a top-level key, to apply to all jobs in the workflow, or within specific jobs.
When you add the permissions key within a specific job, all actions and run commands within that job that use the `GITHUB_TOKEN` gain the access rights you specify.
```

File: README.md [hunk 6]
```diff
@@ -18,6 +18,27 @@ In order to use this action, you will need to generate a JSON file using the fol
...
+## Permissions
+
+It is possible you have a `GITHUB_TOKEN` with restricted access by default, in which case you might get the following error:
+
+```
+Error: GitHubApiUnauthorizedError: Unable to create a check, please make sure that the provided 'repo-token' has write permissions to 'your/repo' - cause: HttpError: Resource not accessible by integration
+```
+
+You need to provide the `checks` write permission:
+```yaml
```
Comment (ggrossetie) [inline README.md:30]:
```suggestion
You need to provide the `checks` write permission:

```yaml
```

**Comment by ggrossetie** [comment]
@francorbacho That's great, thank you! I made tiny updates. Thanks again for taking the time to help others 🤗

**Pull request merged**